bulk_grades/api.py

```diff
@@ -437,6 +437,38 @@ def get_rows_to_export(self):
                     row[f'grade-{block_id}'] = effective_grade
             yield row
 
+    def filtered_column_headers(self):
+        """
+        To trim down grade exports, only show subsections which were modified in a bulk update.
+        Returns: a filtered list of columns to export, preserving modified and non-subsection columns
+        """
+        unmodified_subsections = set()
+        for subsection in self._subsections.keys():
+            unmodified_subsections.add(subsection)
+
+        for row in self.result_data:
+            # Ignore rows which didn't introduce changes
+            if row['status'] == 'No Action':
+                continue
+
+            modified_by_row = set()
+
+            # Get changes this row introduced for an as-yet-unmodified subsection
+            for subsection in unmodified_subsections:
+                if row[f'new_override-{subsection}'] != '':
```

Comment:
would this include a row / column that has just whitespace, or at this point has it been processed out?

Reply:
I believe [it's been pre-processed out](https://github.com/edx/edx-bulk-grades/blob/efbc5f7b28c2c494ff7b7372823c56a101eea14a/bulk_grades/api.py#L350) or, at least, testing it manually by hitting it with a bunch of whitespace seems to function appropriately.

Reply:
^ Correction, it is not actually pre-processed out, I mistook `"No Action"`s for correctly ignoring whitespace.